src/ast.rs

```diff
@@ -1,18 +1,79 @@
+use std::fmt::Display;
+
 use crate::{ByteString, TokenKind};
 
 pub type Block = Vec<Statement>;
 pub type Program = Block;
 
+#[derive(Debug, Eq, PartialEq, Clone)]
+pub enum TryBlockCaughtType {
+    Identifier(ByteString),
+    Union(Vec<ByteString>),
+}
+
 #[derive(Debug, Eq, PartialEq, Clone)]
 pub enum Type {
-    Plain(ByteString),
+    Identifier(ByteString),
     Nullable(ByteString),
-    Union(Vec<ByteString>),
-    Intersection(Vec<ByteString>),
+    Union(Box<Vec<Type>>),
```

Comment:
We don't _need_ to box these vectors do we? Unless you're doing it to reduce memory consumption/size of the enum?

Reply:
indeed, not sure why i added them! ^^"